### ISSUE TITLE
Nullius setting boxed input amount crashes with error about indexing nil

### ISSUE THREAD

**Issue by amacleod** [ClaudeMetz/FactoryPlanner#165]
A helpful fellow player on the Nullius discord taught me that I can right-click the ingredient icon to specify a specific amount of input. When I try to do this to set the input of crushed copper ore boxes to 15, my server crashes and the following error appears in the console log:
```
  67.796 Error MainLoop.cpp:1391: Exception at tick 91154771: The mod Factory Planner (1.1.73) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::Custom event (ID 258)
__factoryplanner__/ui/main/item_boxes.lua:229: attempt to index local 'item' (a nil value)
stack traceback:
        __factoryplanner__/ui/main/item_boxes.lua:229: in function '?'
        __factoryplanner__/ui/dialogs/generic_dialogs.lua:211: in function 'dialog_close'
        __factoryplanner__/ui/base/modal_dialog.lua:187: in function 'exit'
        __factoryplanner__/ui/event_handler.lua:216: in function <__factoryplanner__/ui/event_handler.lua:189>
        [C]: in function 'raise_event'
        __factoryplanner__/util/raise.lua:27: in function 'close_dialog'
        __factoryplanner__/ui/base/modal_dialog.lua:285: in function 'handler'
        __factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
stack traceback:
        [C]: in function 'raise_event'
        __factoryplanner__/util/raise.lua:27: in function 'close_dialog'
        __factoryplanner__/ui/base/modal_dialog.lua:285: in function 'handler'
        __factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
```

![Factorio 1 1 94 2023-11-18 20_23_15](https://github.com/ClaudeMetz/FactoryPlanner/assets/184219/cd8c8a95-2a29-435a-9773-5b1027a178a7)

This happens whether I am using the traditional solver or the matrix solver. I think I can work around the issue by adding the crushing recipe as a sort of "foundation" for my build experimentation.

**Comment by amacleod** [comment]
Looks like this might be a duplicate of https://github.com/ClaudeMetz/FactoryPlanner/issues/158

**Comment by ClaudeMetz** [comment]
It's not actually a duplicate, this is about setting an amount on the top level ingredients, the others were about recipe ingredients. Similar, but not the same :)

Thanks for the report, this is fixed for the next release, due out soon.

**Comment by amacleod** [comment]
Confirmed fixed after loading 1.1.74 into my save. I can correctly set top level ingredients now. Thanks for the fast turnaround, and for the great mod!
